src/gafaelfawr/providers/oidc.py

```diff
@@ -199,6 +211,51 @@ async def logout(self, session: State) -> None:
         """
         pass
 
+    async def _get_ldap_uidNumber(self, uid: str) -> int:
+        """Retrieve the uid number from ldap
+
+        Parameters
+        ----------
+        uid : `str`
+            Username of the user.
+
+        Returns
+        -------
+        uidNumber : `int`
+            The numeric id of the user from LDAP.
+
+        Raises
+        ------
+        gafaelfawr.exceptions.LDAPException
+            The lookup using the uid_number_attr against the LDAP server was not
+            valid (attribute not in LDAP or resultant value not an integer).
+        """
+        assert self._ldap_config.uid_number_attr
+        attr = self._ldap_config.uid_number_attr
+        search = f"(&(Uid={uid}))"
+
+        self._logger.debug(
+            f"querying ldap {self._ldap_config.url} at {self._ldap_config.base_dn} for uid number with {search}"
+        )
+        client = bonsai.LDAPClient(self._ldap_config.url)
+        async with client.connect(is_async=True) as conn:
+            results = await conn.search(
+                self._ldap_config.base_dn,
+                bonsai.LDAPSearchScope.ONE,
+                search,
+                attrlist=[f"{attr}"],
+            )
+            # parse uidNumber
+            uidNumber = None
+            for result in results:
+                try:
+                    self._logger.debug(f"ldap result: {result}")
+                    uidNumber = int(result[attr][0])
```

Comment:
This currently implements getting the last number, but you're using a search of `ONE`, so you should only get one result. Therefore, I think you can just say:
```suggestion
                    return int(result[attr][0])
```

Reply:
good point; i was wondering if returning an error was more appropriate in case ldap returned multiple results.